planner/core/find_best_task.go

```diff
@@ -322,6 +322,20 @@ func getTaskPlanCost(t task, op *physicalOptimizeOp) (float64, bool, error) {
 	case *rootTask:
 		taskType = property.RootTaskType
 	case *copTask: // no need to know whether the task is single-read or double-read, so both CopSingleReadTaskType and CopDoubleReadTaskType are OK
+		if t.(*copTask).indexPlan != nil && t.(*copTask).tablePlan != nil {
```

Comment:
Do we need to use different calculations according to whether `indexPlanFinished` is true or false?

Reply:
updated